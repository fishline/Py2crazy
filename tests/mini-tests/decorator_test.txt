# these weird decorators currently don't work so hto ...
# from Lib/threading.py
@name.setter
def name(self, name):
    pass

# from Lib/csv.py
@fieldnames.setter
def fieldnames(self, value):
    self._fieldnames = value
