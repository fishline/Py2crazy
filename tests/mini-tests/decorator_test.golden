Disassembling top-level module in ./mini-tests/decorator_test.txt
  3       0 LOAD_NAME            @[43m[30mname[39m[49m.setter
          3 LOAD_ATTR            @[43m[30mname.setter[39m[49m
          6 LOAD_CONST           @[43m[30mname.setter[39m[49m
          9 MAKE_FUNCTION        @[43m[30mname.setter[39m[49m
         12 CALL_FUNCTION        @[43m[30mname.setter[39m[49m
         15 STORE_NAME           @[43m[30mname.setter[39m[49m

  8      18 LOAD_NAME            @[43m[30mfieldnames[39m[49m.setter
         21 LOAD_ATTR            @[43m[30mfieldnames.setter[39m[49m
         24 LOAD_CONST           @[43m[30mfieldnames.setter[39m[49m
         27 MAKE_FUNCTION        @[43m[30mfieldnames.setter[39m[49m
         30 CALL_FUNCTION        @[43m[30mfieldnames.setter[39m[49m
         33 STORE_NAME           @[43m[30mfieldnames.setter[39m[49m
         36 LOAD_CONST           @[43m[30mfieldnames.setter[39m[49m
         39 RETURN_VALUE         @[43m[30mfieldnames.setter[39m[49m

Disassembling function name
  5       0 LOAD_CONST               [43m[30mpass[39m[49m
          3 RETURN_VALUE             [43m[30mpass[39m[49m

Disassembling function fieldnames
 10       0 LOAD_FAST                self._fieldnames = [43m[30mvalue[39m[49m
          3 LOAD_FAST                [43m[30mself[39m[49m._fieldnames = value
          6 STORE_ATTR               [43m[30mself._fieldnames[39m[49m = value
          9 LOAD_CONST               [43m[30mself._fieldnames[39m[49m = value
         12 RETURN_VALUE             [43m[30mself._fieldnames[39m[49m = value
